src/renderer/src/renderer/vertexbuffer.cpp

```diff
@@ -0,0 +1,30 @@
+#include "vertexbuffer.h"
+
+#include "platform/opengl/vertexbuffer.h"
+#include "renderer_api.h"
+
+std::unique_ptr<VertexBuffer> VertexBuffer::Create(size_t size) {
+	switch (RendererAPI::instance()->api()) {
+		case RendererAPI::API::None:
+			assert(false);
+			return nullptr;
+		case RendererAPI::API::OpenGL:
+			return std::make_unique<OpenGLVertexBuffer>(size);
+	}
+	assert(false);
+	return nullptr;
+}
+
+std::unique_ptr<VertexBuffer> VertexBuffer::Create(
+		const void* data, size_t size,
+		std::unique_ptr<VertexBufferLayout>&& layout) {
+	switch (RendererAPI::instance()->api()) {
+		case RendererAPI::API::None:
+			assert(false && "Renderer::API::None is not currently supported");
+			return nullptr;
+		case RendererAPI::API::OpenGL:
+			return std::make_unique<OpenGLVertexBuffer>(data, size, std::move(layout));
+	}
+	assert(false);
+	return nullptr;
```

Comment:
this part almost completelly duplicates 4 times, 2 times in this file and 2 times in vertex_array.cpp please unify it

Reply:
Will be in one abstract factory, not in this pr